File: TerminBA/TerminBA.Models/Model/NotificationResponse.cs
using System;

namespace TerminBA.Models.Model
{
    public class NotificationResponse
    {
        public int Id { get; set; }
        public int PostOwnerId { get; set; }
        public int ReservationId { get; set; }
        public ReservationResponse? Reservation { get; set; }
        public string RequesterName { get; set; } = string.Empty;
        public string FacilityName { get; set; } = string.Empty;
        public DateTime Date { get; set; }
        public bool IsSeen { get; set; }
        public string? Reason { get; set; }
        public string Type { get; set; } = string.Empty; // "Cancelation" or "Update"
    }
}

File: TerminBA/TerminBA.UI/terminba_mobile/lib/model/notification_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:terminba_mobile/model/reservation_response.dart';

part 'notification_response.g.dart';

@JsonSerializable()
class NotificationResponse {
  final int id;
  final int postOwnerId;
  final int reservationId;
  final ReservationResponse? reservation;
  final String requesterName;
  final String facilityName;
  final String date;
  final bool isSeen;
  final String? reason;
  final String type;

  NotificationResponse({
    required this.id,
    required this.postOwnerId,
    required this.reservationId,
    this.reservation,
    required this.requesterName,
    required this.facilityName,
    required this.date,
    required this.isSeen,
    this.reason,
    required this.type,
  });

  factory NotificationResponse.fromJson(Map<String, dynamic> json) =>
      _$NotificationResponseFromJson(json);

  Map<String, dynamic> toJson() => _$NotificationResponseToJson(this);
}
